python/cudf/cudf/io/kafka.py

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+from cudf.utils import ioutils
+
+
+@ioutils.doc_kafka_datasource()
+class KafkaDatasource(object):
```

Comment:
We should define some type of `DataSource` base class and inherit from it for this class.

The class could be a Cython class where we hold onto the C datasource via a smart pointer for lifetime management, and then in the io code above we can have one general codepath for handling all custom datasources where it's just something like `return source_info(<datasource *>src[0].c_datasource.get())`